src/server/pps/server/api_server.go

```diff
@@ -2558,28 +2566,30 @@ func (a *apiServer) deletePipeline(ctx context.Context, request *pps.DeletePipel
 		return errors.Wrapf(err, "collection.Delete")
 	}
 
-	if request.KeepRepo {
-		// Remove branch provenance from output and meta
-		if err := pachClient.CreateBranch(
-			request.Pipeline.Name,
-			pipelineInfo.Details.OutputBranch,
-			"",
-			"",
-			nil,
-		); err != nil {
-			return err
-		}
-		if _, err := pachClient.PfsAPIClient.CreateBranch(pachClient.Ctx(), &pfs.CreateBranchRequest{
-			Branch: client.
-				NewSystemRepo(request.Pipeline.Name, pfs.MetaRepoType).
-				NewBranch(pipelineInfo.Details.OutputBranch),
-		}); err != nil && !errutil.IsNotFoundError(err) {
-			return grpcutil.ScrubGRPC(err)
-		}
-	} else {
-		// delete the pipeline's output repo
-		if err := pachClient.DeleteRepo(request.Pipeline.Name, request.Force); err != nil {
-			return err
+	if !missingRepo {
+		if !request.KeepRepo {
+			// delete the pipeline's output repo
+			if _, err := a.env.PfsServer().DeleteRepo(ctx, &pfs.DeleteRepoRequest{
+				Repo:  client.NewRepo(request.Pipeline.Name),
+				Force: request.Force,
+			}); err != nil && !errutil.IsNotFoundError(err) {
+				return errors.Wrap(err, "error deleting pipeline repo")
+			}
+		} else if pipelineInfo.Details != nil {
```

Comment:
should we add a final else that throws an error?

Reply:
Good idea